streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsKafkaClient.java

```diff
@@ -357,55 +351,7 @@ public MetadataResponse fetchMetadata() {
             throw new StreamsException("Inconsistent response type for internal topic metadata request. " +
                 "Expected MetadataResponse but received " + clientResponse.responseBody().getClass().getName());
         }
-        final MetadataResponse metadataResponse = (MetadataResponse) clientResponse.responseBody();
-        return metadataResponse;
-    }
-
-    /**
-     * Check if the used brokers have version 0.10.1.x or higher.
-     * <p>
-     * Note, for <em>pre</em> 0.10.x brokers the broker version cannot be checked and the client will hang and retry
-     * until it {@link StreamsConfig#REQUEST_TIMEOUT_MS_CONFIG times out}.
-     *
-     * @throws BrokerNotFoundException if connecting failed within {@code request.timeout.ms}
-     * @throws TimeoutException if there was no response within {@code request.timeout.ms}
-     * @throws StreamsException if brokers have version 0.10.0.x
-     * @throws StreamsException for any other fatal error
-     */
-    public void checkBrokerCompatibility(final boolean eosEnabled) throws StreamsException {
```

Comment:
What is happening with the broker compatibility check? Has it moved elsewhere?

Reply:
We have decided to remove this check and rely on runtime NetworkClient to throw exception (in the KIP-220 discussion thread). The main reasons are discussed there.

Reply:
Oh - yep - i remember seeing that. thanks